tests/test_encode.py

```diff
@@ -121,7 +121,7 @@ def assert_rgb_rotate(self, input_, is_dash=False):
     self.assertEqual(stream.format.name, "yuv420p")
     self.assertEqual(stream.format.width, WIDTH)
     self.assertEqual(stream.format.height, HEIGHT)
-    self.assertEqual(stream.rate, None)
+    self.assertEqual(stream.codec_context.rate, None)
```

Comment:
No, please update all tests to explicitly test both stream.rate and stream.codec_context.rate

Reply:
The original line seems to have been added in #910 and I don't know if it is even necessary?

Reply:
It is definitely necessary since apparently our test suite did not capture the behaviour you are proposing to fix!

Reply:
What I mean is the previous assertion that it is None doesn't seem to make sense. Same with the updated assertion.
We should be asserting that stream.rate is Fraction(24,1), and I can make that change. I can add assertions on the codec_context too (there seems to be just one other place), but note that we don't do the same for the audio streams currently.

Reply:
> It is definitely necessary since apparently our test suite did not capture the behaviour you are proposing to fix!

Just asserting stream.rate is Fraction(24,1) alone does capture the behavior...

Reply:
Yikes, added that test and hit the problem from #881